### PR TITLE
Move the shadow/integration-test to <CSDK>/integration-test/shadow

### PR DESCRIPTION
**Pull request by andysun2015**
*Description of changes:*
Migrate the integration test of shadow lib and make changes for submoduling shadow library

- Migrate shadow integration test from libraries/aws/shadow/integration-test to integration-test/shadow
- Modify the demos/shadow/shadow_demo_main/CMakeLists.txt for the changes from separate shadow repo
- Make shadow library consistent with the separate repo to pass build check.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: integration-test/shadow/CMakeLists.txt [hunk 1]
```diff
@@ -66,3 +68,28 @@ create_test(${stest_name}
...
+        ${stest_name} PRIVATE
+        CLIENT_CERT_PATH="${CLIENT_CERT_PATH}"
+    )
+endif()
+if(CLIENT_PRIVATE_KEY_PATH)
+    target_compile_definitions(
+        ${stest_name} PRIVATE
+        CLIENT_PRIVATE_KEY_PATH="${CLIENT_PRIVATE_KEY_PATH}"
+    )
+endif()
```
Comment (leegeth) [inline integration-test/shadow/CMakeLists.txt:95]: Nit:please add a newline to the end of file.

Comment (sarenameas) [reply integration-test/shadow/CMakeLists.txt:95]: thanks

**Review by sarenameas — APPROVED** [review]

**Review by leegeth — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Pull request merged**